gcloud-java-compute/src/main/java/com/google/gcloud/compute/ComputeImpl.java

```diff
@@ -1161,6 +1201,124 @@ public Operation deprecate(ImageId image,
     }
   }
 
+  @Override
+  public Disk get(final DiskId diskId, DiskOption... options) {
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      com.google.api.services.compute.model.Disk answer =
+          runWithRetries(new Callable<com.google.api.services.compute.model.Disk>() {
+            @Override
+            public com.google.api.services.compute.model.Disk call() {
+              return computeRpc.getDisk(diskId.zone(), diskId.disk(), optionsMap);
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER);
+      return answer == null ? null : Disk.fromPb(this, answer);
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  @Override
+  public Operation create(final DiskInfo disk, OperationOption... options) {
+    final com.google.api.services.compute.model.Disk diskPb =
+        disk.setProjectId(options().projectId()).toPb();
+    final Map<ComputeRpc.Option, ?> optionsMap = optionMap(options);
+    try {
+      return Operation.fromPb(this,
+          runWithRetries(new Callable<com.google.api.services.compute.model.Operation>() {
+            @Override
+            public com.google.api.services.compute.model.Operation call() {
+              return computeRpc.createDisk(disk.diskId().zone(), diskPb, optionsMap);
+            }
+          }, options().retryParams(), EXCEPTION_HANDLER));
+    } catch (RetryHelper.RetryHelperException e) {
+      throw ComputeException.translateAndThrow(e);
+    }
+  }
+
+  private static Function<com.google.api.services.compute.model.Disk, Disk> diskFromPb(
```

Comment:
Just curious, any reason to put this method here rather than having a package-scope static `Function` in the Disk class (like in Storage's Blob?).  If a user is using a lot of paging and list calls (perhaps an uncommon use case), could it be slightly more efficient to have the `Function` declared once statically in Disk rather than returning a new `Function` each time?  I don't feel strongly about changing it, I'm fine with keeping it as is if you prefer that.